### ISSUE TITLE
X-axis tick mark at first of the month is incorrectly injected into charts with weekly time grain

### ISSUE THREAD

**Issue by fikriabror** [apache/superset#19506]
#### How to reproduce the bug

1. Create a time series charts
2. Click on Time columns is contain the date and time grain is weekly
3. X-axis Weekly time grain not Match with the Marker

### Expected results

The X-axis match with the marker which in this case start from Monday.

### Actual results

The X-axis just appear on the random days.

#### Screenshots

<img width="1431" alt="Screen Shot 2022-04-04 at 16 37 50" src="https://user-images.githubusercontent.com/11493699/161518762-99c6d1c7-a768-41cd-9a55-6107e54a34a7.png">

### Environment

(please complete the following information):

- browser type and version: Chrome
- superset version: `1.0.0`

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [V] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [V] I have reproduced the issue with at least the latest released version of superset.
- [V] I have checked the issue tracker for the same issue and I haven't found one similar.

**Comment by Deckel** [comment]
Do we know if this will be picked up at some point, easily one of the most visible bugs in any dashboard.

**Comment by chitxxx** [comment]
Upvoting. Experiencing similar issues.

**Comment by a-bartoszek** [comment]
I have this same issue in version 2.1.0.  Our "Time Column" field has weekbeginningdate (which are Mondays for us) for a period of a year, and it adds tick marks for the 1st of each month and then for every 7 days after that regardless of the values in our time dimention.  Interestingly, we noticed that this is happening on both the line chart and area chart, HOWEVER, if we pick the lagacy version of each of these charts then we at least get back to weekly tick marks but it uses Sunday dates for these.

**Comment by Khrol** [comment]
Looks like we need to use `alignWithLabel` property from here https://apache.github.io/echarts-handbook/en/concepts/axis/

```
      axisTick: {
        alignWithLabel: true
      },
```

**Comment by rusackas** [comment]
Closing this for a few reasons:
1) It's gone stale - almost 5 months since the last comment
2) The linked PR doesn't look like it'll be merging anytime soon
3) The issue itself is reported on Superset 1.0, and we don't support releases that old (4.0 just came out!)
4) That chart (and much of the UI) is on the path toward deprecation... I'm not sure this is happening in the latest echarts (at least I'm not able to figure out how to reproduce it).

In short, I can't replicate this, on current Superset, and definitely not on the (now very old) 1.0 version. If it's still happening, I'd recommend opening a new issue with updated context and a reproducible example using example/generated data.
